keyboards/sofle/keymaps/default/keymap.c

```diff
@@ -0,0 +1,369 @@
+#include QMK_KEYBOARD_H
+
+enum sofle_layers {
+    /* _M_XYZ = Mac Os, _W_XYZ = Win/Linux */
+    _M_QWERTY,
+    _W_QWERTY,
+    _M_COLEMAK,
+    _W_COLEMAK,
+    _M_LOWER,
+    _W_LOWER,
+    _M_RAISE,
+    _W_RAISE,
+    _ADJUST,
+};
+
+enum custom_keycodes {
+    KC_M_QWERTY = SAFE_RANGE,
+    KC_W_QWERTY,
+    KC_M_COLEMAK,
+    KC_W_COLEMAK,
+    KC_M_LOWER,
+    KC_W_LOWER,
+    KC_M_RAISE,
+    KC_W_RAISE,
+    KC_ADJUST,
+};
+
+/* Linux/win variants */
+#define KC_W_UNDO    LCTL(KC_Z)
+#define KC_W_CUT     LCTL(KC_X)
+#define KC_W_COPY    LCTL(KC_C)
+#define KC_W_PASTE   LCTL(KC_V)
+#define KC_W_PRVWD   LCTL(KC_LEFT)
+#define KC_W_NXTWD   LCTL(KC_RIGHT)
+#define KC_W_LSTRT   KC_HOME
+#define KC_W_LEND    KC_END
+#define KC_W_DLINE   LCTL(KC_BSPC)
+
+/* Mac variants */
+#define KC_M_UNDO  LGUI(KC_Z)
+#define KC_M_CUT   LGUI(KC_X)
+#define KC_M_COPY  LGUI(KC_C)
+#define KC_M_PASTE LGUI(KC_V)
+#define KC_M_PRVWD LALT(KC_LEFT)
+#define KC_M_NXTWD LALT(KC_RIGHT)
+#define KC_M_LSTRT LGUI(KC_LEFT)
+#define KC_M_LEND  LGUI(KC_RIGHT)
+#define KC_M_DLINE LGUI(KC_BSPC)
+
+#define KC_____ KC_TRNS
+#define KC_XXXX KC_NO
+#define KC_RESET RESET
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+/*
+ * QWERTY
+ * ,-----------------------------------------.                    ,-----------------------------------------.
+ * |  `   |   1  |   2  |   3  |   4  |   5  |                    |   6  |   7  |   8  |   9  |   0  |  `   |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * | ESC  |   Q  |   W  |   E  |   R  |   T  |                    |   Y  |   U  |   I  |   O  |   P  | Bspc |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * | Tab  |   A  |   S  |   D  |   F  |   G  |-------.    ,-------|   H  |   J  |   K  |   L  |   ;  |  '   |
+ * |------+------+------+------+------+------|       |    |       |------+------+------+------+------+------|
+ * |LShift|   Z  |   X  |   C  |   V  |   B  |-------|    |-------|   N  |   M  |   ,  |   .  |   /  |RShift|
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_M_QWERTY] = LAYOUT_kc( \
```

Comment:
Please convert this to a regular `LAYOUT` macro. 

We actively discourage the use of `LAYOUT_kc` in the default keymap, as it causes issues for newer/inexperienced users, as it requires redefining anything  that doesn't start with `KC_` by default.

Reply:
Removed as suggested. I had no idea it's discouraged.